packages/frontend/src/hooks/useProcessQr.ts

```diff
@@ -149,6 +149,14 @@ export default function useProcessQR() {
 
       const { qr } = parsed
 
+      if (qr.kind === 'backupTooNew') {
+        await openAlertDialog({
+          message: tx('multidevice_receiver_needs_update'),
+        })
+        callback?.()
```

Comment:
Is this some TypeScript magic?

Reply:
Nvm, I looked it up. I'm guessing this is checking if the callback is defined before invoking it.

Reply:
A general comment about this `useProcessQR()` function. It looks like it's only used in one place, which doesn't pass in a callback, so maybe we should just remove all the callback-invoking code?

Reply:
It's used multiple times and sometimes witha callback:
https://github.com/deltachat/deltachat-desktop/blob/e5d32b6a6736d144442c8371339247dcf013877b/packages/frontend/src/components/dialogs/SetupMultiDevice/ReceiveBackupDialog.tsx#L52
or
https://github.com/deltachat/deltachat-desktop/blob/e5d32b6a6736d144442c8371339247dcf013877b/packages/frontend/src/components/dialogs/QrCode.tsx#L239
Which is for closing a dialog etc.